conan/tools/apple/xcodedeps.py

```diff
@@ -105,6 +105,10 @@ class XcodeDeps(object):
         """)
 
     def __init__(self, conanfile):
+        if self.__class__.__name__ in conanfile.generators:
```

Comment:
it's duplicated multiple times, maybe it makes sense to extract code to its own function?

Reply:
I thought that this would take just 2 lines, the ``if`` and then 1 line for the error message, not 3.
So @RubenRBS do whatever you think it is best, I am happy with either way.

Reply:
My only issue with this is the thing we mentioned on friday that it's not obvious where to place such helper function. I've done so in tool's init, but happy to change. Thanks @SSE4 for your suggestion :)